examples/sha256.rs

```diff
@@ -43,16 +42,16 @@ impl<F: LurkField> CoCircuit<F> for Sha256Coprocessor<F> {
     fn synthesize<CS: ConstraintSystem<F>>(
         &self,
         cs: &mut CS,
-        g: &GlobalAllocations<F>,
+        _g: &GlobalAllocations<F>,
         store: &Store<F>,
         _input_exprs: &[AllocatedPtr<F>],
         input_env: &AllocatedPtr<F>,
         input_cont: &AllocatedContPtr<F>,
     ) -> Result<(AllocatedPtr<F>, AllocatedPtr<F>, AllocatedContPtr<F>), SynthesisError> {
-        // // TODO: Maybe fix this
-        let false_bool = Boolean::from(AllocatedBit::alloc(cs.namespace(|| "false"), Some(false))?);
-
-        let preimage = vec![false_bool; self.n * 8];
+        let preimage = (0..self.n * 8)
+            .map(|i| AllocatedBit::alloc(cs.namespace(|| format!("preimage bit {i}")), Some(false)))
+            .map(|b| b.map(Boolean::from))
+            .collect::<Result<Vec<_>, _>>()?;
```

Comment:
Here's what I see:
- The previous version was unconstrained but economically used only a single shared allocation.
- The new version is still unconstrained but allocates many bits.

For unused generality, the move to many allocated bits seems fine (but unnecessary when we know the preimage will be all zeroes).

Either way, we need to constrain however many bits are allocated to definitely be zero (or to correspond to the input if we accept non-zero). Doing this will be easier if it only involves a single bit.